web/src/main/java/org/wso2/testgrid/web/api/TestPlanService.java

```diff
@@ -107,4 +113,32 @@ public Response getTestPlan(@PathParam("id") String id, @QueryParam("require-tes
                     new ErrorResponse.ErrorResponseBuilder().setMessage(msg).build()).build();
         }
     }
+
+    /**
+     * This has the impleme
+     * @param testPlanRequest  {@link TestPlanRequest} that includes the repository and other necessary detials
+     *                                                to create new test plan.
+     * @return A list of available TestPlans.
+     */
+    @POST
+    @Path("/create")
+    @Consumes(MediaType.APPLICATION_JSON)
+    public Response createTestPlan(TestPlanRequest testPlanRequest) {
+        ConfigXmlGenerator configXmlGenerator = new ConfigXmlGenerator();
+        PipelineManager pipelineManager = new PipelineManager();
+        try {
+            String configXml = configXmlGenerator.getConfigXml(testPlanRequest);
+            return Response.status(Response.Status.OK).
```

Comment:
Response status should be '201 Created.'

Reply:
Fixed.